controller/proxy-injector/patch.go

```diff
@@ -72,6 +73,12 @@ func (p *Patch) addPodLabel(label map[string]string) {
 		Path:  patchPathPodLabel,
 		Value: label,
 	})
+
+	p.patchOps = append(p.patchOps, &patchOp{
+		Op:    "add",
+		Path:  patchPathSelectorMatchLabels,
+		Value: label,
+	})
```

Comment:
I'm not sure that we should modify the spec's selector labels. I tried reverting this change, and your branch still fixes the issue that I encountered on GKE. Plus if we do decide to start modifying selector labels, I'd want to make sure that we also update the `linkerd inject` implementation to match.

And those issues aside, based on my reading of kubernetes/kubernetes#50719, I'm pretty sure the right behavior is for us to treat the selector labels as immutable. How would you feel about reverting this change?

Reply:
hmmm... that's interesting. I thought the error is caused by a mismatch between the selector labels and the pod template labels. The k8s docs say,

> Label selector for pods. Existing ReplicaSets whose pods are selected by this will be the ones affected by this deployment. **It must match the pod template's labels.**

I'll need to do some more testing on this.

Reply:
I did some more testing on this by changing the `apiVersion` of the emojivoto deployments to `apps/v1`. It looks like this change isn't required, as you pointed out. Agree with treating the selector labels as immutable for all the reasons pointed out in kubernetes/kubernetes#50808. I have reverted it.